util/file.go

```diff
@@ -972,3 +973,45 @@ func evalRealPathAndInfo(currentPath string) (string, os.FileInfo, error) {
 
 	return realPath, realInfo, nil
 }
+
+// SanitizePath resolves a file path within a base directory, returning the sanitized path or an error if it attempts
+// to access anything outside the base directory.
```

Comment:
Curious as to why you didn't utilize [the new os.Root](https://pkg.go.dev/os@master#Root). I've never used it, but it seems designed for this kind of thing.

Reply:
good suggestion, I updated the code to use new `os.Root`